test_scripts/API/Navigation/SendLocation/020_SendLocation_boundary_by_parameters.lua

```diff
@@ -0,0 +1,271 @@
+    "countryCode",
+    "postalCode",
+    "administrativeArea",
+    "subAdministrativeArea",
+    "locality",
+    "subLocality",
+    "thoroughfare",
+    "subThoroughfare",
+}
+
+local boundValues ={
+    longitudeDegrees = {lower = -180, upper = 180},
+    latitudeDegrees = {lower = -90, upper = 90},
+    locationName = {lower = "a", upper = string500Char},
+    locationDescription = {lower = "a", upper = string500Char},
+    phoneNumber = {lower = "a", upper = string500Char},
+    addressLines = {lower = {"a"}, upper = {string500Char}}
+}
+
+local locationImage = {lower = "a", upper = string255Char}
+
+local deliveryModeEnum = {
+    "PROMPT",
+    "DESTINATION",
+    "QUEUE"
+}
+
+--[[ Local Functions ]]
+local function sendLocation(parameter, value, self)
+    local paramters  = commonFunctions:cloneTable(request_params)
+
+    paramters[parameter] = value
+
+    local cid = self.mobileSession1:SendRPC("SendLocation", paramters)
+
+    paramters.appID = commonSendLocation.getHMIAppId()
+
+    EXPECT_HMICALL("Navigation.SendLocation", paramters)
+    :Do(function(_,data)
+        self.hmiConnection:SendResponse(data.id, data.method, "SUCCESS", {})
+    end)
+
+    self.mobileSession1:ExpectResponse(cid, { success = true, resultCode = "SUCCESS" })
+end
+
+
+local function sendLocationAddress(subParam, value, self)
+    local paramters  = commonFunctions:cloneTable(request_params)
+
+    paramters["address"] = {}
+    paramters.address[subParam] = value
+
+    local cid = self.mobileSession1:SendRPC("SendLocation", paramters)
+
+    paramters.appID = commonSendLocation.getHMIAppId()
+
+    EXPECT_HMICALL("Navigation.SendLocation", paramters)
+    :Do(function(_,data)
+        self.hmiConnection:SendResponse(data.id, data.method, "SUCCESS", {})
+    end)
+    :ValidIf (function(_,data)
+        local ErrorMessage = ""
+        local ValidationStatus = true
+        for _,k in pairs(addressParams) do
+            if data.params.address[k] and not paramters.address[k] then
+                ErrorMessage = ErrorMessage .. k .. "\n"
+                ValidationStatus = false
+            end
+        end
+
+        if ValidationStatus == false then
+            return false, "SDL sent redundant address parameters to HMI:\n" .. ErrorMessage
+        else
+            return true
+        end
+    end)
+
+    self.mobileSession1:ExpectResponse(cid, { success = true, resultCode = "SUCCESS" })
+end
+
+local function sendLocationImage(value, self)
+    local paramters  = commonFunctions:cloneTable(request_params)
+
+    paramters["locationImage"] = {value = value, imageType = "DYNAMIC"}
+
+    local cid = self.mobileSession1:SendRPC("SendLocation", paramters)
+
+    paramters.appID = commonSendLocation.getHMIAppId()
+    local deviceID = commonSendLocation.getDeviceMAC()
+    paramters.locationImage.value = commonSendLocation.getPathToSDL() .. "storage/"
+    .. commonSendLocation.getMobileAppId(1) .. "_" .. deviceID .. "/" ..tostring(value)
+
+
```

Comment:
@GetmanetsIrina Please remove extra new line

Reply:
Done in 4875635ea9ce242266c4f0d7bbaece1731a76faf